test/builder/builders.go

```diff
@@ -469,3 +469,82 @@ func (csrb *CertificateSigningRequestBuilder) Build() (*certv1.CertificateSignin
 	}
 	return csr, nil
 }
+
+// K8sInstallerConfigBuilder holds the variables and objects required to build an infrastructurev1beta1.K8sInstallerConfig
+type K8sInstallerConfigBuilder struct {
+	namespace    string
+	name         string
+	clusterLabel string
+	byomachine   *infrastructurev1beta1.ByoMachine
+	bundleType   string
+	bundleRepo   string
+}
+
+// K8sInstallerConfig returns a K8sInstallerConfigBuilder with the given name and namespace
+func K8sInstallerConfig(namespace, name string) *K8sInstallerConfigBuilder {
+	return &K8sInstallerConfigBuilder{
+		namespace: namespace,
+		name:      name,
+	}
+}
+
+// WithClusterLabel adds the passed cluster label to the ByoMachineBuilder
+func (b *K8sInstallerConfigBuilder) WithClusterLabel(clusterName string) *K8sInstallerConfigBuilder {
+	b.clusterLabel = clusterName
+	return b
+}
+
+// WithOwnerByoMachine adds the passed Owner ByoMachine to the K8sInstallerConfigBuilder
+func (b *K8sInstallerConfigBuilder) WithOwnerByoMachine(byomachine *infrastructurev1beta1.ByoMachine) *K8sInstallerConfigBuilder {
+	b.byomachine = byomachine
+	return b
+}
+
+// WithBundleRepo adds the passed bundleRepo to the K8sInstallerConfigBuilder
+func (b *K8sInstallerConfigBuilder) WithBundleRepo(bundleRepo string) *K8sInstallerConfigBuilder {
+	b.bundleRepo = bundleRepo
+	return b
+}
+
+// WithBundleType adds the passed bundleType to the K8sInstallerConfigBuilder
+func (b *K8sInstallerConfigBuilder) WithBundleType(bundleType string) *K8sInstallerConfigBuilder {
+	b.bundleType = bundleType
+	return b
+}
+
+// Build returns a K8sInstallerConfig with the attributes added to the K8sInstallerConfigBuilder
+func (b *K8sInstallerConfigBuilder) Build() *infrastructurev1beta1.K8sInstallerConfig {
+	k8sinstallerconfig := &infrastructurev1beta1.K8sInstallerConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "K8sInstallerConfig",
+			APIVersion: infrastructurev1beta1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: b.name,
```

Comment:
nit: It should be Name

Reply:
or we could have a new function called WithGenerateName as the `K8sInstallerConfig ` function's comment says with given name

Reply:
Yup, we can create a new method to set specific name to the resource in builder.
Here in test cases, we need to create a new K8sInstallerConfig resource for each test cases and that's why setting the GenerateName So that it won't conflict with the existing created resource.

Reply:
Added `WIthName()` method in builder